<launch>
  <arg name="max_vel" value="1.5" />
  <arg name="max_acc" value="1.0" />
  <arg name="dynamic" value="0" />

  <!-- main node -->
  <node pkg="plan_manage" name="dyn_planner_node" type="dyn_planner_node" output="screen">
  <!-- remap -->
    <remap from="/laser_cloud_surround" to="/random_forest/all_map"/>
    <remap from="/odom_world" to="/state_ukf/odom"/>

  <!-- sdf map updating -->
    <param name="sdf_map/origin_x" value="-20.0" type="double"/>
    <param name="sdf_map/origin_y" value="-20.0" type="double"/>
    <param name="sdf_map/origin_z" value="-2.0" type="double"/>

    <param name="sdf_map/map_size_x" value="40.0" type="double"/>
    <param name="sdf_map/map_size_y" value="40.0" type="double"/>
    <param name="sdf_map/map_size_z" value="5.0" type="double"/>

    <param name="sdf_map/resolution_sdf" value="0.2" type="double"/>
    <param name="sdf_map/ceil_height" value="2.5" type="double"/>
    <param name="sdf_map/update_rate" value="2.0" type="double"/>
    <param name="sdf_map/update_range" value="8.0" type="double"/>
    <param name="sdf_map/inflate" value="0.1" type="double"/>
    <param name="sdf_map/radius_ignore" value="1.0" type="double"/>

  <!-- planning fsm -->
    <param name="fsm/flight_type" value="1" type="int"/>
    <param name="fsm/thresh_replan" value="3.0" type="double"/>
    <param name="fsm/thresh_no_replan" value="3.0" type="double"/>

    <param name="fsm/wp_num" value="3" type="int"/>

    <param name="fsm/wp0_x" value="8.0" type="double"/>
    <param name="fsm/wp0_y" value="8.0" type="double"/>
    <param name="fsm/wp0_z" value="1.0" type="double"/>

    <param name="fsm/wp1_x" value="-8.0" type="double"/>
    <param name="fsm/wp1_y" value="8.0" type="double"/>
    <param name="fsm/wp1_z" value="1.0" type="double"/>

    <param name="fsm/wp2_x" value="-8.0" type="double"/>
    <param name="fsm/wp2_y" value="-8.0" type="double"/>
    <param name="fsm/wp2_z" value="1.0" type="double"/>

  <!-- planner manager -->
    <param name="manager/time_sample" value="0.5" type="double"/>
    <param name="manager/max_vel" value="$(arg max_vel)" type="double"/>
    <param name="manager/dynamic" value="$(arg dynamic)" type="int"/>
    <param name="manager/margin" value="0.5" type="double"/>

  <!-- path searching -->
    <param name="astar/lambda_heu" value="10.0" type="double"/>
    <param name="astar/resolution_astar" value="0.1" type="double"/>
    <param name="astar/time_resolution" value="0.8" type="double"/>
    <param name="astar/margin" value="0.5" type="double"/>
    <param name="astar/allocate_num" value="100000" type="int"/>

  <!-- path searching -->
    <param name="search/max_tau" value="0.8" type="double"/>
    <param name="search/init_max_tau" value="0.8" type="double"/>
    <param name="search/max_vel" value="$(arg max_vel)" type="double"/>
    <param name="search/max_acc" value="$(arg max_acc)" type="double"/>
    <param name="search/w_time" value="15.0" type="double"/>
    <param name="search/horizon" value="9.0" type="double"/>
    <param name="search/lambda_heu" value="5.0" type="double"/>
    <param name="search/resolution_astar" value="0.1" type="double"/>
    <param name="search/time_resolution" value="0.8" type="double"/>
    <param name="search/margin" value="0.5" type="double"/>
    <param name="search/allocate_num" value="100000" type="int"/>
    <param name="search/check_num" value="5" type="int"/>
    <param name="search/number_of_paths" value="6" type="int"/>
    <param name="search/rrt_avoidance_dist" value="0.5" type="double"/>
    <param name="search/lqr_feasibility_max_vel" value="0.15" type="double"/>
    <param name="search/lqr_min_dis" value="0.6" type="double"/>
    <param name="search/lqr_min_dt" value="0.15" type="double"/>

  <!-- trajectory optimization -->
    <param name="optimization/lamda1" value="10.0" type="double"/>
    <param name="optimization/lamda2" value="0.6" type="double"/>
    <param name="optimization/lamda3" value="0.00001" type="double"/>
    <param name="optimization/lamda4" value="0.001" type="double"/>
    <param name="optimization/dist0" value="0.8" type="double"/>
    <param name="optimization/dist1" value="0.0" type="double"/>
    <param name="optimization/max_vel" value="$(arg max_vel)" type="double"/>
    <param name="optimization/max_acc" value="$(arg max_acc)" type="double"/>

    <param name="optimization/algorithm" value="11" type="int"/>
    <param name="optimization/max_iteration_num" value="100" type="int"/>
    <param name="optimization/order" value="3" type="int"/>

    <param name="bspline/limit_vel" value="$(arg max_vel)" type="double"/>
    <param name="bspline/limit_acc" value="$(arg max_acc)" type="double"/>
    <param name="bspline/limit_ratio" value="1.1" type="double"/>
  </node>

  <!-- trajectory server -->
  <node pkg="plan_manage" name="traj_server" type="traj_server" output="log">
      <remap from="/odom_world" to="/state_ukf/odom"/>
      <remap from="/position_cmd" to="planning/pos_cmd"/>
  </node>

  <!-- use simulator -->
  <include file="$(find plan_manage)/launch/simulator.xml">
  </include>

</launch>